libs/components/src/lib/breadcrumbs/base/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { LinkProps } from '@deriv-com/utils';
+import CustomLink from '../../link';
+import { BreadcrumbsProps } from '../types';
+import { useWindowWidth } from '@react-hook/window-size';
+import { Fragment, useEffect, useState } from 'react';
+import { qtMerge } from '@deriv/quill-design';
+
+export function Base({ size = 'sm', links = [], className }: BreadcrumbsProps) {
+  const [renderLinks, setRenderLinks] = useState<LinkProps[]>([]);
+  const [dropdownLinks, setDropdownLinks] = useState<LinkProps[]>([]);
+
+  const isLastItem = (key: number) => renderLinks.length - 1 > key;
+  const linksLen = links.length;
+  const maxLinks = 3;
+  const hasExtra = useWindowWidth() < 768 && maxLinks < linksLen;
```

Comment:
Can't take out this bunch they are dependent to a state, only `maxLinks` are movable.

Reply:
I meant the `maxLinks` man, we can move that out. thanks 🙏